File: train.js
/*
TASK M: 

Shunday function yozing, u raqamlardan tashkil topgan array qabul qilsin va array ichidagi har bir raqam uchun raqamni ozi va hamda osha raqamni kvadratidan tashkil topgan object hosil qilib, hosil bolgan objectlarni array ichida qaytarsin.
MASALAN: getSquareNumbers([1, 2, 3]) return [{number: 1, square: 1}, {number: 2, square: 4}, {number: 3, square: 9}];
*/
function getSquareNumbers(arr) {
	let result = []

	for (let item of arr) {
		console.log(`number:${item}, square: ${item * item}`)
		result.push({ number: item, square: item * item })
	}

	return result
}

console.log(getSquareNumbers([1, 2, 3, 4]))

// TASK L:

// Shunday function yozing, u string qabul qilsin va string ichidagi hamma sozlarni chappasiga yozib va sozlar ketma-ketligini buzmasdan stringni qaytarsin.
// MASALAN: reverseSentence("we like coding!") return "ew ekil gnidoc";

// function reverseSentence(str) {
// 	const words = str.split(' ')
// 	let newSentence = ''

// 	for (let word of words) {
// 		let reversedWord = word.split('').reverse().join('')
// 		// Boshlanishida bo'sh joy qo'shilmasligi uchun:
// 		newSentence = newSentence ? newSentence + ' ' + reversedWord : reversedWord
// 	}

// 	return newSentence
// }

// const result = reverseSentence('we like coding')
// console.log(result) // "ew ekil gnidoc"

File: train.py
'''
TASK M: 

Shunday function yozing, u raqamlardan tashkil topgan array qabul qilsin va array ichidagi har bir raqam uchun raqamni ozi va hamda osha raqamni kvadratidan tashkil topgan object hosil qilib, hosil bolgan objectlarni array ichida qaytarsin.
MASALAN: getSquareNumbers([1, 2, 3]) return [{number: 1, square: 1}, {number: 2, square: 4}, {number: 3, square: 9}];
'''
# def getSquareNumbers(arr):
#     result = []
#     for item in arr:
#         a = f"number:{item}, square: {item*item}"
#         print(a)
#         result.append({a})

#     return result


# print(getSquareNumbers([1, 2, 3,]))

'''TASK N:

Shunday function yozing, u string qabul qilsin va string palindrom yani togri oqilganda ham, orqasidan oqilganda ham bir hil oqiladigan soz ekanligini aniqlab boolean qiymat qaytarsin.

MASALAN: palindromCheck("dad") return true;  palindromCheck("son") return false;'''

def palindromCheck(str):
    a = str[:len(str)]
    b = a[::-1]

    # print(a, b)

    if a == b:
        return True
    else:
        return False


print(palindromCheck("dad")) # True
print(palindromCheck("son")) # False